Beta version of retry mechanism (#35)
Othoz/gcsfs: CHANGELOG.rst

Unreleased
----------

> Added

Author: ```suggestion
Changed
```

> Changed

Author: ```suggestion
1.4.0 - 12.06.2020
------------------

Changed
```

'''''

> - Experimental: Clients now automatically retry GCS requests that return with "503 Service Unavailable" or "504 Gateway Timeout.

Reviewer: - Experimental: the underlying HTTP client is now configured to automatically retry requests that return a status code of "429 Too Many Requests", "502 Bad Gateway", "503 Service Unavailable" and "504 Gateway Timeout".

> Added
> '''''
> - Experimental: Clients now automatically retry GCS requests that return with "503 Service Unavailable" or "504 Gateway Timeout.

Author: ```suggestion
Changed
'''''''
- The underlying HTTP client is now configured to automatically retry requests that return a status code of "429 Too Many Requests", "502 Bad Gateway", "503 Service Unavailable" and "504 Gateway Timeout".
```
